### PR TITLE
added linter files

### PR DESCRIPTION
**Pull request by Ambrosegithub**
### Added Link to Live Version

- Deploy App online 
- Added video demo link
- Added linter file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
 The BUDGETARY APP is a web application, where we manage our expenses.
```
Comment (SajjadAhmad14) [inline README.md]: Please use [this](https://github.com/microverseinc/readme-template) professional README tempalate frm Microverse.

File: app/models/user.rb [hunk 2]
```diff
@@ -2,11 +2,10 @@ class User < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
```
Comment (SajjadAhmad14) [inline app/models/user.rb]: Please style Sign Up`, and `Sign In` pages as per the design page. They have no style currenly.

User when clicks on `Sign up` button there is an error and app does not route the user to categories page. Kindly see the error below.

sign_up_error

File: app/views/groups/index.html.erb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <div id ="group-index-title">
```
Comment (SajjadAhmad14) [inline app/views/groups/index.html.erb:1]: The categories page is not designed keeping the view of the orgignal design provided. For example three buttons are layed out in a way that they are user friendly and do not not follow user friendly UI design principle. Kindly revisit the design page and try to copy the same colors, layout, and fonts.

btn_categories

categories

File: app/views/expenses/index.json.jbuilder [hunk 4]
```diff
@@ -1 +1 @@
-json.array! @expenses, partial: "expenses/expense", as: :expense
```
Comment (SajjadAhmad14) [inline app/views/expenses/index.json.jbuilder]: Please improve the design of transactions page as per the design guidelines. Kindly follow the same layout, fonts, and colors as defined in the orignal design page.
transac

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -44,55 +44,53 @@
   # triggering implicit auto-inclusion in groups with matching metadata.
```
Comment (SajjadAhmad14) [inline spec/spec_helper.rb]: Please add integration specs for the app using rspec and Capybara.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Ambrose, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: README.md [hunk 6]
```diff
@@ -1,65 +1,126 @@
-The BUDGETARY APP is a web application, where we manage our expenses.
+# BUDGET APP
```
Comment (FrederickMih) [inline README.md:1]: - [ ] Hi @Ambrosegithub, the previous reviewer pointed out to update your README.md file using the new template. Kindly do the adjustments.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @Ambrosegithub, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

# Highlights :green_circle: 

- All user inputs have been validated :+1: 


## Suggested changes

- [ ] Initially, I got the console error below on the screenshot below when I tried to sign up.


Screenshot from 2022-11-26 05-27-19

- [ ] As requested by the previous reviewer, kindly make sure that your design matches the one seen in the [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=).
- colors should be the same
- Typography: font face, size, and weight,
- Layout: composition and space between elements should be exactly the same as in the [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=)

Your design seems to be still different from that proposed in the guidelines. Your app should be responsive. All call-to-action buttons should be the same with the same color as seen on the screenshot below:

| Yours | Design guidelines |
| --- | --- |
| Screenshot from 2022-11-26 05-57-46 | Screenshot from 2022-11-26 05-58-29 |







_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: app/views/splash/index.html.erb [hunk 7]
```diff
@@ -1,6 +1,12 @@
 <div class="splash-container">
    <h1 class="splash-title">BUDGETARY APP TRANSACTIONS</h1>
+   <div class ="sign-in">
+  <%= link_to "Sign in", new_user_session_path, method: :delete, class:"nav-link1" %>
+
+  <%= link_to "Sign up", new_user_registration_path, method: :delete, class:"nav-link2" %>
```
Comment (rloterh) [inline app/views/splash/index.html.erb]: - [ ] Sign up a new user throws an error, when the signup form is submitted. As mentioned in the previous reviews kindly ensure to improve upon the same.

**Sign up error:**
image

File: app/views/splash/index.html.erb [hunk 8]
```diff
@@ -1,6 +1,12 @@
 <div class="splash-container">
    <h1 class="splash-title">BUDGETARY APP TRANSACTIONS</h1>
+   <div class ="sign-in">
+  <%= link_to "Sign in", new_user_session_path, method: :delete, class:"nav-link1" %>
+
```
Comment (rloterh) [inline app/views/splash/index.html.erb]: Again, kindly ensure the Login screen, and the general design of your app is consistent with the [template design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=&), just as mentioned in previous reviews.
- [ ] Layout of each screen should match the original design. Items should be properly positioned, similar to the original design
- [ ] Contents should properly fit on mobile screen versions
- [ ] Ensure that typography (fonts) used matches the original design
- [ ] Background colors of items in a layout should match the original design
- [ ] A navigation button to move to and fro the screens should be present

File: README.md [hunk 9]
```diff
@@ -1,65 +1,126 @@
...
+
+### Usage
+--Once you have your repository cloned on your machine, feel free to modify it as needed.
+- To add your own page
+
+### Testing with Rspec and Capybara
+  - In your terminal run > 
+     bundle exec rspec
+  - run rspec spec
+
```
Comment (rloterh) [inline README.md:84]: - [ ] OPTIONAL
Kindly ensure to add the _Future Features_ section to your ReadMe, which should include 1-3 potential features to be added

File: README.md [hunk 10]
```diff
@@ -1,65 +1,126 @@
...
+  
+## Authors
```
Comment (rloterh) [inline README.md]: - [ ] OPTIONAL
I think you should change the header title _Authors_ to _Author_ in your ReadMe, as this project is worked on by a sole contributor. You can optionally also change it to _Author(s)_ to cater for future contributors.

File: README.md [hunk 11]
```diff
@@ -1,65 +1,126 @@
...
-Give a ⭐️ if you like this project!
+##### [Answer_2]No because its on their local database .
 
-Acknowledgments
-Credits go to Gregoire at Behance
-📝 License
+## 📝 License
 This project is MIT licensed.
 
+NOTE: we recommend using the MIT license - you can set it up quickly by using templates available on GitHub. You can also use any other license if you wish.
```
Comment (rloterh) [inline README.md]: - [ ] OPTIONAL
Kindly endeavor to remove the guide note below the MIT license.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Ambrose, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Properly added required models, controllers, and corresponding views
- Devise correctly implemented for authentication
- Schema follow given ERD
- App is running as intended

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly check the comments marked **[OPTIONAL]** under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 6 above)
Comment (rloterh) [reply README.md:1]: well resolved! ✔️

File: app/views/splash/index.html.erb (hunk 8 above)
Comment (Ambrosegithub) [reply app/views/splash/index.html.erb]: Done

File: README.md (hunk 10 above)
Comment (Ambrosegithub) [reply README.md]: Done

File: README.md (hunk 11 above)
Comment (Ambrosegithub) [reply README.md]: Done

File: app/views/splash/index.html.erb (hunk 7 above)
Comment (Ambrosegithub) [reply app/views/splash/index.html.erb]: Done

File: app/views/splash/index.html.erb (hunk 8 above)
Comment (emmanuellekamwa) [reply app/views/splash/index.html.erb]: - [ ] Your app design still doesn't respect the [app design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=&). Please kindly make sure it respects the design.

**Review by emmanuellekamwa — CHANGES_REQUESTED** [review]
**STATUS: CHANGES REQUESTED ⛔🛑❌ **
Hello @Ambrosegithub 👋
Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

## Highlights

- You documented your work professionally.✔
- You asked for the changes in a polite and friendly way.

 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — CHANGES_REQUESTED** [review]
### Hi @Ambrosegithub  :hand: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

### To highlight! :+1: 
- Professional readme file ✔️
- No linter errors ✔️
almoo


 ## Status: Required Changes ♻️
- [ ] Kindly push your last changes to be reflected in your deployed application.
- [ ] `node_modules` should not be pushed on guthub to make your project clean and easy to load. Please put it inside `.gitignore` before pushing on github.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by Davidosky007 — APPROVED** [review]
Hi @Ambrosegithub 👍, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 👍 
* well Written PR description 👍 
## Optional suggestions
* I strongly suggest you fix these errors i get when i run your tests 
* Also these commands can help you remove nodule modules from GithubGitHub 
image

> git rm -r --cached node_modules

> git commit -am "node_modules be gone!"
> 
> git push origin master

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**